compiler/src/dotty/tools/dotc/core/tasty/TastyPickler.scala

```diff
@@ -74,17 +80,19 @@ class TastyPickler(val rootCls: ClassSymbol) {
 
   /** Returns a non-cryptographic 64-bit hash of the array.
    *
-   *  from https://en.wikipedia.org/wiki/PJW_hash_function#Implementation
+   *  from https://en.wikipedia.org/wiki/PJW_hash_function#Algorithm
    */
   private def pjwHash64(data: Array[Byte]): Long = {
     var h = 0L
-    var high = 0L
     var i = 0
     while (i < data.length) {
-      h = (h << 4) + data(i)
-      high = h & 0xF0000000L
-      h ^= high >> 24
-      h &= ~high
+      val d = data(i) & 0xFFL // Interpret byte as unsigned byte
+      h = (h << 8) + d
+      val high = h & 0xFF00000000000000L
+      if (high != 0) {
```

Comment:
Remove the if, it's not necessary and branches should be avoided in tight loops.

Reply:
Done